static/js/apps/nl_interface/app.tsx

```diff
@@ -28,6 +29,38 @@ export function App(): JSX.Element {
   const [queries, setQueries] = useState<string[]>([]);
   const [contextList, setContextList] = useState<any[]>([]);
 
+  // Executes a single query through the search box.
+  function executePrompt(prompt: string) {
+    if (prompt) {
+      (
+        document.getElementById("query-search-input") as HTMLInputElement
+      ).value = prompt;
+      (
+        document.getElementById("rich-search-button") as HTMLButtonElement
+      ).click();
```

Comment:
this feels hacky, but fine for now!

Maybe there is some way to make use of the initialValue prop in TextSearchBar instead of having to find the element and set the value

Reply:
I tried that but was unsuccessful... I also didn't want to mess with the state too much. This somewhat simulates what a user would do :) but I will add a TODO to do this in a less hacky way.